sljit_src/sljitProtExecAllocator.c

```diff
@@ -118,12 +119,17 @@ static SLJIT_INLINE int create_tempfile(void)
 	size_t tmp_name_len;
 	char *dir;
 	size_t len;
+	mode_t mode;
 
 #ifdef HAVE_MEMFD_CREATE
 	/* this is a GNU extension, make sure to use -D_GNU_SOURCE */
 	fd = memfd_create("sljit", MFD_CLOEXEC);
-	if (fd != -1)
-		return fd;
+	if (fd != -1) {
+		if (fchmod(fd, 0) == 0)
+			return fd;
+		else
```

Comment:
No need else here.

Reply:
OK, but I was expecting that in the extremely unlikely case when fchmod could fail here it was still probably better to fallback to use a different way to get the tmpfile than continue using one file that might be possible to hijack externally IMHO